### PR TITLE
`ultralytics 8.3.183` New NDJSON dataset format support

### PR DESCRIPTION
**Pull request by glenn-jocher**
## 🛠️ PR Summary

<sub>Made with ❤️ by [Ultralytics Actions](https://github.com/ultralytics/actions)<sub>

### 🌟 Summary
Adds first-class NDJSON dataset support for YOLO11, including auto-conversion during training, comprehensive docs, and tests — making large and cloud-hosted datasets easier to use. 🚀

### 📊 Key Changes
- NDJSON dataset support
  - Trainer now detects `.ndjson` inputs and auto-converts them to standard YOLO format before training (async + parallel). ⚙️
  - New async converter `convert_ndjson_to_yolo` downloads remote images, writes labels, and generates `data.yaml`.
  - Supports annotations for detection, segmentation, pose, OBB, and classification in the NDJSON spec.
- Documentation updates
  - New “Ultralytics NDJSON format” section with examples, usage (Python/CLI), and advantages.
  - Improved YAML examples across detect/segment/pose/OBB docs using clearer “example” blocks.
  - Anchor fix: “Usage” -> “Usage Example” for accurate cross-references.
- Testing
  - Adds an online test to train with NDJSON (`coco8-ndjson.ndjson`).
- Misc
  - Version bump to 8.3.183.
  - Adds @ShuaiLYU to docs authors.

### 🎯 Purpose & Impact
- Simpler dataset management: Single-file datasets that are easy to version, share, and maintain. 📦
- Better scalability: Stream large datasets line-by-line without loading everything into memory. 🧠
- Cloud-friendly: Train directly from remote image URLs — great for cloud and Ultralytics HUB workflows. ☁️
- Faster onboarding: Auto-conversion means you can start training with a single `.ndjson` path:
  - CLI: `yolo detect train data=path/to/dataset.ndjson model=yolo11n.pt`
  - Python:
    ```python
    from ultralytics import YOLO
    model = YOLO("yolo11n.pt")
    model.train(data="path/to/dataset.ndjson", epochs=100, imgsz=640)
    ```
- Note: Current auto-conversion in the trainer is wired through the detection dataset check; classification NDJSON may not auto-route yet. See the Ultralytics Docs for details. 📘

### CODE REVIEW

**Comment by UltralyticsAssistant** [comment]
👋 Hello @glenn-jocher, thank you for submitting an `ultralytics/ultralytics` 🚀 PR! This is an automated response to help speed things up—an Ultralytics engineer will also review and assist you shortly.

- ✅ **Define a Purpose**: Clearly explain the purpose of your fix or feature in your PR description, and link to any [relevant issues](https://github.com/ultralytics/ultralytics/issues). Ensure your commit messages are clear, concise, and adhere to the project's conventions.
- ✅ **Synchronize with Source**: Confirm your PR is synchronized with the `ultralytics/ultralytics` `main` branch. If it's behind, update it by clicking the 'Update branch' button or by running `git pull` and `git merge main` locally.
- ✅ **Ensure CI Checks Pass**: Verify all Ultralytics [Continuous Integration (CI)](https://docs.ultralytics.com/help/CI/) checks are passing. If any checks fail, please address the issues.
- ✅ **Update Documentation**: Update the relevant [documentation](https://docs.ultralytics.com/) for any new or modified features.
- ✅ **Add Tests**: If applicable, include or update tests to cover your changes, and confirm that all tests are passing.
- ✅ **Sign the CLA**: Please ensure you have signed our [Contributor License Agreement](https://docs.ultralytics.com/help/CLA/) if this is your first Ultralytics PR by writing "I have read the CLA Document and I sign the CLA" in a new message.
- ✅ **Minimize Changes**: Limit your changes to the **minimum** necessary for your bug fix or feature addition. _"It is not daily increase but daily decrease, hack away the unessential. The closer to the source, the less wastage there is."_ — Bruce Lee

PR-specific notes for “Add NDJSON training”:
- Great addition bringing NDJSON-to-YOLO11 conversion plus trainer integration 🎉 Please ensure:
  - Unit tests cover `convert_ndjson_to_yolo(...)` (including splits, names mapping, empty annotations, and URL download errors) and the `Trainer.get_dataset()` NDJSON path.
  - New dependencies (`requests`, `tqdm`) and imports are handled consistently and don’t introduce optional-import pitfalls.
  - Parallel downloader behavior is robust: timeouts, retries, and thread-local session lifecycle are considered. Logging on failures looks good—confirm it’s visible but not noisy.
  - Documentation includes the new converter reference and a minimal training snippet so users can `model.train(data="*.ndjson")` end-to-end.
  - Data YAML generation aligns with existing expectations for downstream tasks and preserves compatibility across detect/segment/pose/classify where applicable.
- If there are any known edge cases (e.g., mixed annotation types per image, missing URLs, or unusual file names), please call them out in the PR description for reviewers 🔎

For more guidance, please refer to our [Contributing Guide](https://docs.ultralytics.com/help/contributing/). Don’t hesitate to leave a comment if you have any questions. Thank you for contributing to Ultralytics! 🚀

**Comment by glenn-jocher** [comment]
@ultralytics/platform PR for NDJSON training

**Comment by glenn-jocher** [comment]
## ⚡ Actions Trigger

<sub>Made with ❤️ by [Ultralytics Actions](https://www.ultralytics.com/actions)<sub>

GitHub Actions below triggered via workflow dispatch on this PR branch `ndjson` at 2025-08-20 18:37:21 UTC with `@ultralytics/run-ci` command:

* ✅ [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/runs/17107227640): `ci.yml` run 10269
* ✅ [Publish Docker Images](https://github.com/ultralytics/ultralytics/actions/runs/17107227729): `docker.yml` run 954

**Comment by glenn-jocher** [comment]
@ultralytics/yolo new NDJSON format export to YOLO for @ultralytics/platform

**Comment by UltralyticsAssistant** [comment]
Huge thanks and congrats on the merge, @glenn-jocher! “The best way to predict the future is to invent it.” — Alan Kay. Your first-class NDJSON support for YOLO11 does exactly that: it turns single-file, cloud-friendly datasets into a plug-and-train experience with async auto-conversion, robust docs, and tests. This makes large, remote datasets effortless for users and unlocks smoother Ultralytics HUB workflows. Quick win for the community: `yolo detect train data=path/to/dataset.ndjson model=yolo11n.pt`. See the Ultralytics Docs for details: https://docs.ultralytics.com. Thank you for pushing the standard forward!
